python/vyos/ifconfig/interface.py

```diff
@@ -898,6 +899,60 @@ def add_to_bridge(self, bridge_dict):
             # set bridge port path priority
             if 'priority' in bridge_config:
                 self.set_path_cost(bridge_config['priority'])
+            
+            vlan_filter = 0
+            
+            vlan_del = set()
+            vlan_add = set()
+
+            if 'native_vlan' in bridge_config:
+                vlan_filter = 1
+                cmd = f'bridge vlan del dev {self.ifname} vid 1'
+                self._cmd(cmd)
+                vlan_id = bridge_config['native_vlan']
+                if int(vlan_id) != 1:
+                    vlan_del.add(1)
+                cmd = f'bridge vlan add dev {self.ifname} vid {vlan_id} pvid untagged master'
+                self._cmd(cmd)
+                vlan_add.add(vlan_id)
+
+            if 'allowed_vlan' in bridge_config:
+                vlan_filter = 1
+
+            if vlan_filter:
```

Comment:
Why is this `if vlan_filter` required? Just executing the commands below under `if 'allowed_vlan' in bridge_config:` have the same effect.